src/mlpack/core/tree/vantage_point_tree/vantage_point_tree_impl.hpp

```diff
@@ -0,0 +1,921 @@
+inline bool VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+                            SplitType>::IsLeaf() const
+{
+  return !left;
+}
+
+/**
+ * Returns the number of children in this node.
+ */
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename BoundMetricType, typename...> class BoundType,
+         template<typename SplitBoundType, typename SplitMatType>
+             class SplitType>
+inline size_t VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+                              SplitType>::NumChildren() const
+{
+  if (left && right)
+    return 2;
+  if (left)
+    return 1;
+
+  return 0;
+}
+
+/**
+ * Return a bound on the furthest point in the node from the center.  This
+ * returns 0 unless the node is a leaf.
+ */
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename BoundMetricType, typename...> class BoundType,
+         template<typename SplitBoundType, typename SplitMatType>
+             class SplitType>
+inline
+typename VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::ElemType
+VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::FurthestPointDistance() const
+{
+  if (!IsLeaf())
+    return 0.0;
+
+  // Otherwise return the distance from the center to a corner of the bound.
+  return 0.5 * bound.Diameter();
+}
+
+/**
+ * Return the furthest possible descendant distance.  This returns the maximum
+ * distance from the center to the edge of the bound and not the empirical
+ * quantity which is the actual furthest descendant distance.  So the actual
+ * furthest descendant distance may be less than what this method returns (but
+ * it will never be greater than this).
+ */
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename BoundMetricType, typename...> class BoundType,
+         template<typename SplitBoundType, typename SplitMatType>
+             class SplitType>
+inline
+typename VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::ElemType
+VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::FurthestDescendantDistance() const
+{
+  return furthestDescendantDistance;
+}
+
+//! Return the minimum distance from the center to any bound edge.
+template<typename MetricType,
+         typename StatisticType,
+         typename MatType,
+         template<typename BoundMetricType, typename...> class BoundType,
+         template<typename SplitBoundType, typename SplitMatType>
+             class SplitType>
+inline
+typename VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::ElemType
+VantagePointTree<MetricType, StatisticType, MatType, BoundType,
+    SplitType>::MinimumBoundDistance() const
+{
+  return bound.MinWidth() / 2.0;
```

Comment:
In this case I think this will just be `bound.OuterRadius()`. (That's the same as this, I think, just without a division.)